File: Check If All 1's Are at Least Length K Places Away/Check If All 1's Are at Least Length K Places Away.java
// Solution for Check If All 1's Are at Least Length K Places Away in JAVA

class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int prev = -1;
        for (int curr = 0; curr < nums.length; curr++) {
            if (nums[curr] == 0) continue;
            if (prev != -1 && curr - prev - 1 < k) {
                return false;
            }
            prev = curr;
        }
        return true;
    }
}

File: Check If All 1's Are at Least Length K Places Away/Check If All 1's Are at Least Length K Places Away.py
# Solution for Check If All 1's Are at Least Length K Places Away in PY

class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        prev = -1
        for curr in range(len(nums)):
            if nums[curr] == 0:
                continue
            if prev != -1 and curr - prev - 1 < k:
                return False
            prev = curr
        return True
